### PR TITLE
Support custom job descriptions in .mono_repo.yml files

### PR DESCRIPTION
**Pull request by jakemac53**
Adds custom names for jobs on travis. By default the name looks like:

```
SDK: <version> - DIR: <directory> - TASKS: <all task commands being ran>
```

The part after `TASKS` is configurable via an extra key `description` on the job config.

### CODE REVIEW

**Comment by kevmoo** [comment]
https://travis-ci.org/dart-lang/mono_repo/jobs/408581253#L560

That's one I haven't seen before...

**Comment by jakemac53** [comment]
> https://travis-ci.org/dart-lang/mono_repo/jobs/408581253#L560
> 
> That's one I haven't seen before...

whoah ya that is weird

File: .travis.yml [hunk 1]
```diff
@@ -4,14 +4,17 @@ language: dart
 jobs:
   include:
     - stage: smoke_test
+      name: "SDK: dev - DIR: mono_repo - TASKS: dartfmt && dartanalyzer"
```
[the remark this answers is not in the document]
Comment (jakemac53) [reply .travis.yml]: Sure, but I don't know if its really worth the complexity to support?

File: mono_repo/.mono_repo.yml [hunk 2]
```diff
@@ -4,7 +4,8 @@ dart:
 
 stages:
   - smoke_test:
-    - group:
```
[the remark this answers is not in the document]
Comment (jakemac53) [reply mono_repo/.mono_repo.yml]: See the last task here https://travis-ci.org/dart-lang/mono_repo/builds/408697627, basically just toList the commands from the tasks

File: mono_repo/lib/src/mono_config.dart [hunk 3]
```diff
@@ -52,6 +52,10 @@ class MonoConfig {
 
 @JsonSerializable()
 class TravisJob extends Object with _$TravisJobSerializerMixin {
+  @override
```
[the remark this answers is not in the document]
Comment (jakemac53) [reply mono_repo/lib/src/mono_config.dart]: Sure, but later ;)

File: mono_repo/.mono_repo.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (kevmoo) [reply mono_repo/.mono_repo.yml]: ack

File: .travis.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kevmoo) [reply .travis.yml]: not no...if I get bored...

**Pull request merged**